### PR TITLE
Do not requeue messages that violate a constraint

### PR DESCRIPTION
**Pull request by kaizimmerm**
Reviewers: @SirWayne @michahirsch 
Signed-off-by: kaizimmerm <kai.zimmermann@bosch-si.com>

### CODE REVIEW

**Comment by michahirsch** [comment]
:+1:

File: hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/DelayedRequeueExceptionStrategy.java [hunk 1]
```diff
@@ -59,11 +61,18 @@ protected boolean isUserCauseFatal(final Throwable cause) {
     }
 
     private boolean invalidMessage(final Throwable cause) {
-        return doesNotExist(cause) || cause instanceof TooManyStatusEntriesException
-                || cause instanceof InvalidTargetAddressException || cause instanceof ToManyAttributeEntriesException;
+        return doesNotExist(cause) || quotaHit(cause) || invalidContent(cause);
```
Comment (SirWayne) [inline hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/DelayedRequeueExceptionStrategy.java]: After a small discussion with @michahirsch , we believe it's better to add the MessageHandlingException

Comment (SirWayne) [inline hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/DelayedRequeueExceptionStrategy.java]: Found another one `org.springframework.amqp.support.converter.MessageConversionException`

Comment (SirWayne) [inline hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/DelayedRequeueExceptionStrategy.java]: Please also add 'org.springframework.amqp.support.converter.MessageConversionException' e.g. for invalid json

**Comment by kaizimmerm** [comment]
done

**Pull request merged**